components/nav.js

```diff
@@ -66,17 +142,125 @@ export const NavButton = ({ color = 'red', sx, ...props }) => (
         boxShadow: '0 0 0 2px',
         outline: 'none'
       },
-      ...sx
+      ...csx
     }}
   />
 )
 
+const IconWrapper = ({ csx, children, ...props }) => {
+  return (
+    <motion.div
+      {...props}
+      sx={{
+        display: 'inline-flex',
+        padding: '4px',
+        color: 'red',
+        borderRadius: 'circle',
+        textDecoration: 'none',
+        mr: [3, 4],
+        cursor: 'pointer',
+        transition: 'box-shadow .125s ease-in-out',
+        ':hover,:focus': {
+          boxShadow: '0 0 0 2px',
+          outline: 'none'
+        },
+        ...csx
+      }}
+    >
+      {children}
+    </motion.div>
+  )
+}
+
+const SearchBar = ({ setVisible, visible, search, ...props }) => {
+  const inp = useRef(null)
+
+  const handle_focus = () => {
+    const display_val = inp.current.style.display
+
+    if (display_val == 'none') inp.current.blur()
+    else inp.current.focus()
+  }
+
+  const onClick = () => {
+    setVisible(!visible)
+  }
+
+  const retAnim = visible => {
+    if (visible) {
+      console.log(window.innerWidth <= 512 ? 'mobVisible' : 'visible')
```

Comment:
might want to remove this

Reply:
Hey, yeah I was using this statement for some debugging, lemme remove that :)

Thanks, you are awesome 🙌

Reply:
Anytime